### PR TITLE
E1.33 cherry pick v

### PR DESCRIPTION
**Pull request by peternewman**

### CODE REVIEW

File: libs/acn/BrokerConnectReplyInflator.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            << NetworkToHost(pdu_data.e133_version);
+
+  BrokerConnectReply reply(pdu_data.connection_code,
+                           NetworkToHost(pdu_data.e133_version),
+                           UID(pdu_data.broker_uid),
+                           UID(pdu_data.client_uid));
+
+  if (m_broker_connect_reply_handler.get()) {
+    m_broker_connect_reply_handler->Run(headers,
+                                        reply);
```
Comment (kripton) [inline libs/acn/BrokerConnectReplyInflator.cpp:69]: why did you break line here?

File: libs/acn/E133StatusHelper.cpp [hunk 2]
```diff
@@ -142,5 +142,104 @@ string ConnectStatusCodeToString(E133ConnectStatusCode connect_status_code) {
   }
   return "Unknown E1.33 Connect Status Code";
 }
+
+
+bool IntToRPTStatusCode(uint16_t input,
+                        RPTStatusVector *rpt_status_code) {
+  switch (input) {
```
Comment (kripton) [inline libs/acn/E133StatusHelper.cpp:149]: rpt_status_code could also use a null check here

File: libs/acn/E133StatusHelper.cpp [hunk 3]
```diff
@@ -142,5 +142,104 @@ string ConnectStatusCodeToString(E133ConnectStatusCode connect_status_code) {
...
+bool RPTStatusCodeToRDMStatusCode(RPTStatusVector rpt_status_code,
+                                  RDMStatusCode *rdm_status_code) {
+  // TODO(Peter): Fill in the gaps, possibly adding additional RDMStatusCodes
+  // if required
+  switch (rpt_status_code) {
+//    case ola::acn::VECTOR_RPT_STATUS_UNKNOWN_RPT_UID:
+//      *rdm_status_code = ola::rdm::;
+//      return true;
+    case ola::acn::VECTOR_RPT_STATUS_RDM_TIMEOUT:
+      *rdm_status_code = ola::rdm::RDM_TIMEOUT;
```
Comment (kripton) [inline libs/acn/E133StatusHelper.cpp:217]: same: no null check before de-reference

**Review by kripton — APPROVED** [review]
see inline comments

File: libs/acn/BrokerClientEntryRPTPDU.cpp [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+
+/*
+ * Pack the header into a buffer.
+ */
+void BrokerClientEntryRPTPDU::PackHeader(OutputStream *stream) const {
+  BrokerClientEntryHeader::broker_client_entry_pdu_header header;
+  m_header.ClientCid().Pack(header.client_cid);
+  stream->Write(
```
[the remark this answers is not in the document]
Comment (peternewman) [reply libs/acn/BrokerClientEntryRPTPDU.cpp:108]: As per the previous, for now I'll park this, PDU::Pack which is the only code that calls PackHeader/PackData has protection now. I guess we can add it to ALL PackHeader/PackData in one go if we deem it useful.

File: libs/acn/BrokerClientEntryUpdateInflator.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * BrokerClientEntryUpdateInflator.h
+ * Interface for the BrokerClientEntryUpdateInflator class.
+ * Copyright (C) 2023 Peter Newman
+ */
+
+#ifndef LIBS_ACN_BROKERCLIENTEENTRYUPDATEINFLATOR_H_
```
[the remark this answers is not in the document]
Comment (peternewman) [reply libs/acn/BrokerClientEntryUpdateInflator.h]: I was confused why cpplint hadn't noticed, but it has now and also fixed now.

File: libs/acn/E133StatusHelper.cpp (hunk 2 above)
Comment (peternewman) [reply libs/acn/E133StatusHelper.cpp:166]: Tests and null check added.

File: libs/acn/E133StatusHelper.cpp (hunk 3 above)
Comment (peternewman) [reply libs/acn/E133StatusHelper.cpp:240]: Tests and null check added.

File: libs/acn/BrokerConnectReplyInflator.cpp (hunk 1 above)
Comment (peternewman) [reply libs/acn/BrokerConnectReplyInflator.cpp:69]: I'm not sure, probably a copy/paste lack of tidying from another function, or maybe adding/removing stuff in the callback. Now fixed anyway.

File: libs/acn/BrokerClientEntryRPTPDU.cpp [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+using ola::acn::CID;
+using ola::io::OutputStream;
+using ola::network::HostToNetwork;
+using ola::rdm::UID;
+
+/*
+ * Size of the header portion.
+ */
+unsigned int BrokerClientEntryRPTPDU::HeaderSize() const {
```
[the remark this answers is not in the document]
Comment (peternewman) [reply libs/acn/BrokerClientEntryRPTPDU.cpp:41]: I'll push it to the main PR as if we were to do it, we should do it for all PDUs.

I assume there shouldn't be any knock-ons with the inheritance etc?

File: include/ola/e133/E133Enums.h [hunk 7]
```diff
@@ -24,6 +24,12 @@
 namespace ola {
 namespace e133 {
 
+// Appendix A - Endpoints and Table 3-1: Endpoint ID Allocation
+static const uint16_t NULL_ENDPOINT = 0x0000;
```
[the remark this answers is not in the document]
Comment (peternewman) [reply include/ola/e133/E133Enums.h:30]: Without this we've got slightly more Constants.h than Enums.h. Most of them seem to have a mix of both in.

I'll put the rename in the future column just so I don't waste lots of timing fighting merge conflicts for now.

**Comment by peternewman** [comment]
It's finally actually passing the tests too! Please take a final look @kripton !

**Review by kripton — APPROVED** [review]
Re-review done

**Comment by kripton** [comment]
... and the compilation checks also look much better now

**Pull request merged**